### PR TITLE
Remove the Coarse provider

### PR DESCRIPTION
**Pull request by ldorau**
### Description

Remove the Coarse provider.

Fixes: #900

Requires:
- [x] #931 
- [x] #932 

### Checklist

- [x] Code compiles without errors locally
- [x] All tests pass locally
- [x] CI workflows execute properly

### CODE REVIEW

File: test/disjointPoolFileProv.cpp [hunk 1]
```diff
@@ -0,0 +1,363 @@
...
+};
+
+INSTANTIATE_TEST_SUITE_P(
+    FileWithMemoryStrategyTest, FileWithMemoryStrategyTest,
+    ::testing::Values(UMF_COARSE_MEMORY_STRATEGY_FASTEST,
+                      UMF_COARSE_MEMORY_STRATEGY_FASTEST_BUT_ONE,
+                      UMF_COARSE_MEMORY_STRATEGY_CHECK_ALL_SIZE));
+
+TEST_P(FileWithMemoryStrategyTest, disjointFileMallocPool_simple1) {
+    umf_memory_provider_handle_t malloc_memory_provider;
```
Comment (KFilipek) [inline test/disjointPoolFileProv.cpp:43]: Some declarations are assigned to NULL, others not. Can it make Coverity scream about using variables without assignment?

File: test/disjointPoolFileProv.cpp [hunk 2]
```diff
@@ -0,0 +1,363 @@
...
+    umf_memory_pool_handle_t pool;
+    umf_result =
+        umfPoolCreate(umfDisjointPoolOps(), file_memory_provider,
+                      disjoint_pool_params, UMF_POOL_CREATE_FLAG_NONE, &pool);
+    ASSERT_EQ(umf_result, UMF_RESULT_SUCCESS);
+    ASSERT_NE(pool, nullptr);
+
+    umf_result = umfDisjointPoolParamsDestroy(disjoint_pool_params);
+
+    umf_memory_provider_handle_t prov = NULL;
```
Comment (KFilipek) [inline test/disjointPoolFileProv.cpp:92]: In C++ files, `nullptr` is preferred for any pointer type.

Comment (ldorau) [reply test/disjointPoolFileProv.cpp]: Done

File: test/disjointPoolFileProv.cpp (hunk 1 above)
Comment (ldorau) [reply test/disjointPoolFileProv.cpp]: I cannot predict what Coverity can detect. 
I have initialized all `malloc_memory_provider` to `nullptr`.
Done

**Review by KFilipek — APPROVED** [review]
LGTM

**Comment by ldorau** [comment]
@bratpiorka @vinser52 please review

**Comment by ldorau** [comment]
Lines: +400 / −3136  :-)

**Pull request merged**